sqlglot/expressions.py

```diff
@@ -1117,3 +1570,141 @@ def _all_functions():
 
 
 ALL_FUNCTIONS = _all_functions()
+
+
+def _maybe_parse(
+    code_or_expression,
+    *,
+    expression_type=None,
+    dialect=None,
+    prefix=None,
+    parser_opts=None,
+):
+    if isinstance(code_or_expression, Expression):
+        return code_or_expression
+
+    parser_opts = parser_opts or {}
+    code = str(code_or_expression)
+    if prefix:
+        code = f"{prefix} {code}"
+
+    from sqlglot.dialects import Dialect
+
+    dialect = Dialect.get_or_raise(dialect)()
+
+    if expression_type:
+        result = dialect.parse_into(expression_type, code, **parser_opts)
+    else:
+        result = dialect.parse(code, **parser_opts)
+
+    if not result:
+        return None
+    return result[0]
+
+
+T = TypeVar("T", bound=Expression)
```

Comment:
the other type hints are self documenting, these are on private methods, is this necessary?

you know my stance on types. but i can probably compromise on the ones you have in the ABC or the __init__ methods since those are public. but is this particular type necessary?

Reply:
Ah, good catch. These aren't necessary anymore.